tests/functional/metadata/test_update.py

```diff
@@ -0,0 +1,181 @@
+        LOGGER.info("Adding artifacts")
+        for count in range(0, int(10)):
+            filename = f"test/{names_generator.generate_name()}-{count}.tar.gz"
+            target = {
+                "info": {
+                    "length": 54321,
+                    "hashes": {
+                        "blake2b-256": (
+                            "716f6e863f744b9ac22c97ec7b76ea5f5908bc5b2f67c6151"
+                            "0bfc4751384ea7a"
+                        )
+                    },
+                },
+                "path": filename,
+            }
+            targets.append(target)
+
+        result = http_request(
+            "POST",
+            url="/api/v1/artifacts",
+            json={"targets": targets},
+        )
+        data_result = result.json()["data"]
+        LOGGER.info(f"Added task_id: {data_result['task_id']}")
+        pytest.rstuf_added_targets.append(result.json())
+        time.sleep(0.5)
+
+    LOGGER.info(
+        "Stop adding artifacts. "
+        f"Total requests: {len(pytest.rstuf_added_targets )}"
+    )
+
+
+@scenario(
+    "../../features/metadata/update.feature",
+    "Updating Root metadata full signed",
+)
+def test_updating_root_metadata_full_signed():
+    """Updating Root metadata full signed."""
+
+
+@given("RSTUF is running and operational")
+def running_multiple_requests():
+    pass
+
+
+@then(
+    "the RSTUF is receiving multiple requests",
+    target_fixture="send_rstuf_requests",
+)
+def rstuf_receiving_requests(http_request):
+    stop_requests = threading.Event()
+    thread = threading.Thread(
+        target=rstuf_requests, args=(stop_requests, http_request)
+    )
+    thread.start()
+    return stop_requests
+
+
+@when(
+    "the RSTUF key holders send a full signed metadata",
+    target_fixture="response",
+)
+def send_signed_update_metadata(send_rstuf_requests, http_request):
+    # Start thread sending requests to add artifacts
+    pytest.rstuf_thread = send_rstuf_requests
+    # Wait some requests in the broker before run metadata update
+    time.sleep(2)
+    LOGGER.info("[METADATA UPDATE] Submiting Root Metadata Update")
+    with open("metadata-update-payload.json") as f:
+        paylaod_json = json.loads(f.read())
+
+    result = http_request(
+        "POST",
+        url="/api/v1/metadata",
+        json=paylaod_json,
+    )
+    return result
+
+
+@then(
+    "the API requester should get status code '202' with 'task_id'",
+    target_fixture="task_id",
+)
+def task_is_received(response):
+    assert response.status_code == 202, pytest.rstuf_thread.set()
+
+    json_response = response.json()
+    task_id = json_response["data"]["task_id"]
+    LOGGER.info(f"[METADATA UPDATE]  Metadata Updated by {task_id}")
+    return task_id
+
+
+@then(
+    "the API requester gets from endpoint 'GET /api/v1/task' status 'SUCCESS'"
+)
+def task_is_finished(task_id, http_request):
+    count = 0
+
+    # check for the metadata update task
+    while count < 60:
```

Comment:
The timeout is not part of our BDD test or a behavior condition.
I added it to avoid it getting stuck for some reason (GHA issues).

We need to implement it somehow to make it configurable, and I still need to see the use case.